packages/office-addin-mock/src/officeMockObject.ts

```diff
@@ -0,0 +1,178 @@
+    } else {
+      this[propertyName] = this.properties.get(propertyName)?.value;
+    }
+  }
+
+  private loadCalled() {
+    if (!this.loaded) {
+      this.loaded = true;
+      this.value = `Error, context.sync() was not called`;
+    }
+  }
+
+  private loadMultipleProperties(properties: string) {
+    properties
+      .replace(/\s/g, "")
+      .split(",")
+      .forEach((completeProperties: string) => {
+        this.loadNavigational(completeProperties);
+      });
+  }
+
+  private loadNavigational(completePropertyName: string) {
+    const properties: Array<string> = completePropertyName.split("/");
+    let navigationalOfficeMockObject: OfficeMockObject = this;
+
+    // Iterating through navigational properties
+    for (let i = 0; i < properties.length - 1; i++) {
+      const property = properties[i];
+
+      const checkingUndefined =
+        navigationalOfficeMockObject.properties.get(property);
+      if (checkingUndefined) {
+        navigationalOfficeMockObject = checkingUndefined;
+      } else {
+        throw new Error(
+          `Navigational property ${property} needs to be present in object model before load is called.`
+        );
+      }
+    }
+    const scalarProperty: string = properties[properties.length - 1];
+    navigationalOfficeMockObject.loadScalar(scalarProperty);
+  }
+
+  private loadScalar(scalarPropertyName: string) {
+    if (this.properties.has(scalarPropertyName)) {
+      this.properties.get(scalarPropertyName)?.loadCalled();
+      this.assignValue(scalarPropertyName);
+
+      this.properties
+        .get(scalarPropertyName)
+        ?.properties.forEach((property: OfficeMockObject) => {
+          property.loadCalled();
+        });
+    } else {
+      throw new Error(
+        `Property ${scalarPropertyName} needs to be present in object model before load is called.`
+      );
+    }
+  }
+
+  private populate(objectData: OfficeObject) {
+    try {
+      Object.keys(objectData).forEach((property: string) => {
+        if (typeof objectData[property] === "object") {
+          this.addMock(property);
+          this[property].populate(objectData[property]);
+        } else if (typeof objectData[property] === "function") {
+          this.addMockFunction(property, objectData[property]);
+        } else {
+          this.setMock(property, objectData[property]);
+        }
+      });
+      usageDataObject.reportSuccess("populate()");
+    } catch (err: any) {
+      usageDataObject.reportException("populate()", err);
+    }
+  }
+
+  private resetValue(value: unknown) {
+    this.value = `Error, property was not loaded`;
+    this.valueBeforeLoaded = value;
+    this.loaded = false;
+  }
+
+  private properties: Map<string, OfficeMockObject>;
+  private loaded: boolean;
+  private value: unknown;
+  private valueBeforeLoaded: unknown;
+  private isMockObject: boolean | undefined;
+  /* eslint-disable-next-line */
+  [key: string]: any;
+}
+
+class OfficeObject {
```

Comment:
It represents the Object to be used when populating Office JS with data. It was a suggested way to remove typescript errors, because the following error appears if I just use a normal object (instead of this class):
```
Element implicitly has an 'any' type because expression of type 'string' can't be used to index type 'Object'.
No index signature with a parameter of type 'string' was found on type 'Object'.```

Other way to make code compile is to use a bunch of `as any`. So instead of calling: `objectData[property]`, it would have to call: `(objectData as any)[property]`.

The use of the class is then to make the code cleaner. I will change its name to match more its use.

Reply:
If it's public, add a description to explain what it is.

Reply:
It is not public, but I guess a description can't hurt.